2_process.smk

```diff
@@ -0,0 +1,155 @@
+    # needed to link unzipped files with unzip_archive rule
+    drivers_directory = f'inputs_{lake.group_id}'
+    unzip_archive_drivers = checkpoints.unzip_archive.get(
+        file_category=file_category,
+        directory_name=drivers_directory
+    ).output[0]
+    clarity_directory = f'clarity_{lake.group_id}'
+    unzip_archive_clarity = checkpoints.unzip_archive.get(
+        file_category=file_category,
+        directory_name=clarity_directory
+    ).output[0]
+    ice_flags_directory = f'ice_flags_{lake.group_id}'
+    unzip_archive_ice_flags = checkpoints.unzip_archive.get(
+        file_category=file_category,
+        directory_name=ice_flags_directory
+    ).output[0]
+
+    # dynamic filenames
+    drivers_file = f'{unzip_archive_drivers}/{lake.meteo_filename}'
+    clarity_file = f'{unzip_archive_clarity}/gam_{lake.site_id}_clarity.csv'
+    ice_flags_file = f'{unzip_archive_ice_flags}/pb0_{lake.site_id}_ice_flags.csv'
+    return [drivers_file, clarity_file, ice_flags_file]
+
+
+# Summarize training sequences
+rule process_mntoha:
+    input:
+        mntoha_lake_sequence_files
+    output:
+        "2_process/out/mntoha_sequences/mntoha_sequences_summary.csv"
+    run:
+        save_sequences_summary(input, output[0])
+
+
+# Create .npy of input/output sequences for one lake to use for training and testing
+rule mntoha_lake_sequences:
+    input:
+        "2_process/tmp/mntoha/lake_metadata_augmented.csv",
+        "2_process/tmp/mntoha/temperature_observations_interpolated.csv",
+        lambda wildcards: dynamic_filenames(wildcards.site_id, file_category='dynamic_mntoha')
+    output:
+        "2_process/out/mntoha_sequences/sequences_{site_id}.npy"
+    params:
+        config = config
+    script:
+        "2_process/src/lake_sequences_mntoha.py"
+
+
+# Add elevation to MNTOHA lake metadata
+rule augment_mntoha_lake_metadata:
+    input:
+        "1_fetch/out/lake_metadata.csv"
+    output:
+        "2_process/tmp/mntoha/lake_metadata_augmented.csv"
+    script:
+        "2_process/src/make_lake_metadata_augmented.py"
+
+
+def mntoha_obs_file(wildcards):
+    '''
+    Return MNTOHA temperature observations filepath.
+
+    Depend on unzip_archive checkpoint to ensure that
+    temperature_observations.csv gets unzipped.
+
+    :param wildcards: Snakemake wildcards (empty).
+    :returns: Path of MNTOHA temperature observations csv
+
+    '''
+    # Trigger checkpoint to unzip temperature_observations.csv
+    obs_file_directory = checkpoints.unzip_archive.get(
+        file_category='obs_mntoha',
+        directory_name='temperature_observations'
+    ).output[0]
+    return os.path.join(obs_file_directory, "temperature_observations.csv")
+
+
+# Add column of observation depths interpolated to nearest modeling mesh node
+rule interpolate_mntoha_obs_depths:
+    input:
+        # "1_fetch/out/obs_mntoha/temperature_observations/temperature_observations.csv"
+        mntoha_obs_file
```

Comment:
Okay I had a thought how you could make the use of the unzip checkpoint a bit more modular - assuming this approach would still make `dynamic_filenames()` and `mntoha_obs_file()` depend on `unzip_archive` in `snakemake`'s eyes.

You could add a function to this `.smk` that just calls the checkpoint, like so:
```python
def return_unzipped_archive(file_category, directory_name):
   # Trigger checkpoint to unzip zipped directory
    unzipped_archive = checkpoints.unzip_archive.get(
        file_category=file_category,
        directory_name=directory_name
    ).output[0]
    return(unzipped_archive)
```

Then you could change line 148 to 
```python
lambda wildcards: get_mntoha_obs_file(wildcards, file_category='obs_mntoha', directory_name='temperature_observations', file_name='temperature_observations.csv')
```

And change `mntoha_obs_file()` to 
```python
def get_mntoha_obs_file(wildcards, file_cateogory, directory_name, file_name):
    obs_file_directory = return_unzipped_archive(file_category, directory_name)
    return os.path.join(obs_file_directory, file_name)
```

And you could update line 106 to:
```ptyhon
lambda wildcards: get_dynamic_filenames(wildcards.site_id, file_category='dynamic_mntoha', drivers_dir_template =''inputs_{}", clarity_dir_template="clarity_{}", ice_dir_template="ice_flags_{}")
```
    
and `dynamic_filenames()` to
```python
def get_dynamic_filenames(site_id, file_category = 'dynamic_mntoha'):
    # make this function depend on fetch_mntoha_metadata
    # needed because lake_metadata.csv is used to determine dynamic files
    lake_metadata_file = checkpoints.fetch_mntoha_metadata.get().output[0]
    lake_metadata = pd.read_csv(lake_metadata_file)
    lake = lake_metadata.loc[lake_metadata['site_id']==site_id].iloc[0]
    # also make this function depend on unzip_archive
    # needed to link unzipped files with unzip_archive rule
    drivers_directory = drivers_dir_template.format(lake.group_id)
    unzip_archive_drivers = return_unzipped_archive(file_category, directory_name=drivers_directory)
    clarity_directory = clarity_dir_template.format(lake.group_id)
    unzip_archive_clarity = return_unzipped_archive(file_category, directory_name=clarity_directory)
    ice_flags_directory = ice_dir_template.format(lake.group_id)
    unzip_archive_ice_flags = return_unzipped_archive(file_category, directory_name=ice_flags_directory)

    # dynamic filenames
    drivers_file = f'{unzip_archive_drivers}/{lake.meteo_filename}'
    clarity_file = f'{unzip_archive_clarity}/gam_{lake.site_id}_clarity.csv'
    ice_flags_file = f'{unzip_archive_ice_flags}/pb0_{lake.site_id}_ice_flags.csv'
    return [drivers_file, clarity_file, ice_flags_file]
```

Reply:
Nice! I'll test this out.